### PR TITLE
Release v0.30.2

### PR DESCRIPTION
**Pull request by chukarsten**
# v0.30.2 Aug. 16, 2021
### Fixes
- Updated changelog and version numbers to match the release.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+    * Testing Changes
+
+.. warning::
+
+    **Breaking Changes**
+
+
+**v0.30.2 Aug. 16, 2021**
+    * Fixes
+        * Updated changelog and version numbers to match the release.
```
Comment (dsherry) [inline docs/source/release_notes.rst:17]: Yep. For ultimate clarity please mention that v0.30.1 was erroneously released without a release PR and that v0.30.2 replaces it.

**Review by dsherry — APPROVED** [review]
Left one change request, otherwise approved on green tests.

The deps update tests will fail due to [this outstanding PR](https://github.com/alteryx/evalml/pull/2630) -- we've decided to force-merge this to get the release out and we'll merge the deps update afterwards.

File: docs-requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 docutils>=0.15.2,<0.17
 pydata-sphinx-theme>=0.3.1
+astroid<=2.6.6
```
Comment (dsherry) [inline docs-requirements.txt:3]: If we're gonna do this in the release PR please add a separate log entry mentioning it

**Review by freddyaboulton — APPROVED** [review]
Thank you for your hard work @chukarsten !  🚢

**Review by dsherry — APPROVED** [review]

**Pull request merged**